test/python/circuit/test_extensions_standard.py

```diff
@@ -1486,5 +1488,52 @@ def test_kwarg_deprecation(self, instr_name, inst_class, n_params, kwarg_map):
         self.assertEqual(cargs, [])
 
 
+@ddt
+class TestStandardEquivalenceLibrary(QiskitTestCase):
+    """Standard Extension Test."""
+
+    @data(
+        HGate, CHGate, IGate, RGate, RXGate, CRXGate, RYGate, CRYGate, RZGate,
+        CRZGate, SGate, SdgGate, CSwapGate, TGate, TdgGate, U1Gate, CU1Gate,
+        U2Gate, U3Gate, CU3Gate, XGate, CXGate, CCXGate, YGate, CYGate,
+        ZGate, CZGate
+    )
+    def test_definition_parameters(self, gate_class):
+        """Verify decompositions from standard equivalence library match definitions."""
+
+        n_params = len(set(signature(gate_class.__init__).parameters) - {'label', 'self'})
+        param_vector = ParameterVector('th', n_params)
+        float_vector = [0.1 * i for i in range(n_params)]
+
+        param_gate = gate_class(*param_vector)
+        float_gate = gate_class(*float_vector)
+
+        param_entry = std_eqlib.get_entry(param_gate)
+        float_entry = std_eqlib.get_entry(float_gate)
+
+        if not param_gate.definition:
```

Comment:
since `gate.definition` is going away can you make this test independent of that or not yet?

Reply:
This only checks that the definition and the first decomposition are the same. I'll move it to `test_extensions_standard` and add `Operator.equiv(Gate, decomposition)` tests.